lib/BaseController.js

```diff
@@ -556,9 +567,6 @@ module.exports = class BaseController {
 
         let original = clone(file);
         this.reconcileFields(file, lastApplied);
-        if (objectPath.get(file, ['metadata', 'annotations']) === null) {
-          objectPath.set(file, ['metadata', 'annotations'], {});
-        }
```

Comment:
this would still be necessary since `this.reconcileFields(file, lastApplied);` actually is going through and `nulling` values. So its a guard against trying to write to annotations after reconcileFields nulled it.